pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -388,20 +392,31 @@ func applyUpstreamTLSSettings(cluster *v2.Cluster, tls *networking.TLSSettings)
 					},
 				},
 				ValidationContext: certValidationContext,
+				AlpnProtocols:     ClusterALPNProtocols,
 			},
 			Sni: tls.Sni,
 		}
 	}
+
+	if tls.Mode != networking.TLSSettings_DISABLE {
+		// alpn is set for TLS, so specify h2 configuration.
+		addHTTP2Options(cluster)
+	}
```

Comment:
Compare with istio mTLS. Not any mTLS.

Reply:
All TLS support alpn, why not all tls?
If the other side is mtls http1.1, they would negotiate that.

Reply:
Even the old ones?

Reply:
I mean old servers ?